x-pack/legacy/plugins/ml/server/models/data_visualizer/data_visualizer.ts

```diff
@@ -277,7 +367,7 @@ export class DataVisualizer {
       aggs: buildSamplerAggregation(aggs, samplerShardSize),
     };
 
-    const resp = await this.callWithRequest('search', {
+    const resp = await this.context.core.elasticsearch.dataClient.callAsCurrentUser('search', {
```

Comment:
Should we be using the same approach that @alvarezmelissa87 used in the calendars route and `calendar_manager.ts`?

```
const actualClient = isLegacy === true ? client : client.ml!.mlClient.callAsCurrentUser;
```

Reply:
Yeah, I think we should be using the `mlClient`. 
The `mlClient` is set up in `plugin.ts` with the `plugins` option to extend the elasticsearch client.
```
    const mlClient = core.elasticsearch.createClient('ml', { plugins: [elasticsearchJsPlugin] });
    http.registerRouteHandlerContext('ml', (context, request) => {
      return {
        mlClient: mlClient.asScoped(request),
      };
    });
```

That's the difference between `context.core.elasticsearch.dataClient.callAsCurrentUser` and using `.ml!.mlClient.callAsCurrentUser`. We should always use the mlClient to make sure it's consistent and so we don't have to worry about changing things later if we add more extensions to the elasticsearch client.

Reply:
Changed to `mlClient`